core/sql/optimizer/NATable.cpp

```diff
@@ -3678,9 +3682,9 @@ NAType* getSQColTypeForHive(const char* hiveType, NAMemory* heap)
 
   if ( !strncmp(hiveType, "decimal", 7) )
   {
-    Int16 i=0, pstart=0, pend=0, sstart=0, send=0, p=0, s = 0;
-    Int16 hiveTypeLen = strlen(hiveType);
-    char pstr[16], sstr[16];
+    Lng32 i=0, pstart=0, pend=0, sstart=0, send=0, p=-1, s = -1;
```

Comment:
Just a nit: Since a "long" is a 64 bit number, it is better to use Int32 instead of Lng32. The "Lng32" came from porting 32 bit code, where a "long" meant 32 bits, and the port kept it at 32 bits.